### PR TITLE
Add client libraries list to Newtonsoft.Json compat package README

### PR DESCRIPTION
**Pull request by scottaddie**
Replace the 1st occurrence of the vague "some Azure SDKs" verbiage with a list of applicable Track 2 client libraries. Changes made per an email thread with @AlexGhiondea.

### CODE REVIEW

File: sdk/core/Microsoft.Azure.Core.NewtonsoftJson/README.md [hunk 1]
```diff
@@ -1,11 +1,13 @@
...
 
-The [Azure.Core package][azure_core_package] contains types shared by all Azure SDK client libraries.
-This library contains converters dependent on the [Newtonsoft.Json package][newtonsoft_json_package] for use with Azure.Core.
+The [Azure.Core package][azure_core_package] contains types shared by the latest Azure SDK client libraries. This `Newtonsoft.Json` compatibility library:
 
-## Getting started
+- Contains converters dependent upon the [Newtonsoft.Json][newtonsoft_json_package] package.
+- Enables `Newtonsoft.Json` usage for serializing and deserializing model types from the following client libraries:
+  - [Azure.DigitalTwins.Core][azure_digital_twins_core_package]
+  - [Azure.Search.Documents][azure_search_documents_package]
```
Comment (AlexGhiondea) [inline sdk/core/Microsoft.Azure.Core.NewtonsoftJson/README.md:8]: The model types are the ones that are serialized (so customer data), not types from the Azure libraries.

Comment (scottaddie) [reply sdk/core/Microsoft.Azure.Core.NewtonsoftJson/README.md:8]: If I change the word "from" to "with", does that work for you?

Comment (AlexGhiondea) [reply sdk/core/Microsoft.Azure.Core.NewtonsoftJson/README.md:8]: yes, I think so.

Comment (heaths) [reply sdk/core/Microsoft.Azure.Core.NewtonsoftJson/README.md]: +1. You could may change "model types *from* the following" to something like "model types for use with the following".

Comment (heaths) [reply sdk/core/Microsoft.Azure.Core.NewtonsoftJson/README.md]: I think it's still rather vague and we'll get questions about it not working as expected. For example, recently-ish people were trying to serialize Key Vault SDK models and didn't come out like they expected (we don't support it). If you're changing just "from" to "with", I think you need to say "customer model types".

**Review by AlexGhiondea — APPROVED** [review]

File: sdk/core/Microsoft.Azure.Core.NewtonsoftJson/README.md (hunk 1 above)
Comment (scottaddie) [reply sdk/core/Microsoft.Azure.Core.NewtonsoftJson/README.md]: @heaths Please review the latest changes.

**Review by heaths — APPROVED** [review]
Great!

**Comment by scottaddie** [comment]
Thanks for the help, @heaths & @AlexGhiondea! I'll work with @danieljurek to get this update published.

**Pull request merged**